src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/JsonOutputWriter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.jsonwriter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.output.OutputWriter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class JsonOutputWriter implements OutputWriter {
+
+    private final Map<String, OutputFieldBuilder> fieldBuilders;
+    private final ObjectMapper objectMapper;
+    private static final byte[] BATCH_START = "[\n".getBytes(StandardCharsets.UTF_8);
```

Comment:
@ivanyu I tried to use Jackson Streaming like [here](https://www.baeldung.com/jackson-streaming-api), but the problem was that it required create a one generator per output stream. This gives 2 options:

- recreate generator each time in `writeRecord` function 
- create a generator and keep it till `writeLastRecord` would be called. This add additional problems, beucase output stream in generator and in function input could not match.

So, as a result I keep the most basic streaming-like version with a hardcoded separators.